### ISSUE TITLE
[Request] Separately publish standalone nexus client

### ISSUE THREAD

**Issue by sugarmanz** [gradle-nexus/publish-plugin#148]
I've integrated the `NexusClient` from this Gradle plugin within a stupid simple [CLI](https://github.com/player-ui/rules_player/blob/main/distribution/NexusStagingCli.kt#L19) to be used as part of a [Bazel](https://bazel.build/) release pipeline in order to provide better staging repo automation. It would be really nice if:

1. I could download the JAR from Maven Central instead of needing to [add the Gradle plugin portal as a generic Maven repo](https://github.com/bazel-build-sandbox/main-code-repo/blob/main/WORKSPACE.bazel#L62)
1. I could download a JAR that doesn't contain the Gradle plugin fluff for a non-Gradle use case
1. Maybe even contribute back the CLI portion such that it could be used in a generic sense -- even potentially publishing to `brew` or something to be used in an adhoc manner

I get that this is somewhat out of scope for this org/repo, but you've done a great job building out the `NexusClient` and I would love to be able to bring the goodness that this Gradle plugin provides to the Bazel ecosystem in a more stable, official capacity. Let me know what you think!

**Comment by szpak** [comment]
I'm glad you have found it useful! In general, a separation of the lib is possible, but unfortunately, generates some maintenance overhead, especially related to the releasing. Therefore, it might take some time on our side to perform (also to decide if the lib should shadow the retrofit dependency or plugin should do that), so I propose you to temporarily:
1. (Recommended) Use the published JAR (there is just shadowed retrofit2 which is needed and shadowed failsafe for retrying which you probably handle on your own) or,
2. Copy the client to your codebase.

and verify it works fine for you or report if you need anything to be added to the client.

Once we handle the separation, you should be able to seamlessly switch to the new dependency. What it be feasible for you?

**Comment by sugarmanz** [comment]
Yep! Makes sense to me. I can confirm that we are able to use the published JAR from the Gradle plugin portal just fine. I definitely understand that splitting it out is non-trivial, so no rush, just wanted to express my use case. Please let me know if I can be of any help here!
